src/LazyMessageReader.ts

```diff
@@ -28,8 +28,8 @@ export class LazyMessageReader<T = unknown> {
   }
 
   // Return the size of our message within the buffer
-  size(buffer: Readonly<Uint8Array>): number {
-    const view = new DataView(buffer.buffer, buffer.byteOffset);
+  size(buffer: ArrayBufferView): number {
+    const view = new DataView(buffer.buffer, buffer.byteOffset, buffer.byteLength);
```

Comment:
Not sure if it was intentional that this byteLength was missing – doubt it would have affected much except maybe covering up bugs with the reader if the underlying buffer was longer than the given view's byteLength. The tests still pass this way. 🤷